pages/bib/[id]/index.tsx

```diff
@@ -76,27 +89,62 @@ export default function BibPage({
     isArchiveCollection: bib.isArchiveCollection,
   })
 
-  const refreshItemTable = async (newQuery: BibQueryParams) => {
+  // Load all items via client-side fetch if page is first loaded with viewAllItems prop passed in
+  // Namely, when the page is accessed with the /all route
+  useEffect(() => {
+    if (viewAllItems) void refreshItemTable(query, true)
+    // Disable eslint exhaustive-deps rule because we only want this to run once on page load
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
```

Comment:
Why can't you do this on the server side again? If I land on `/all`, why can't all items be fetched on the server and returned as props to the client? This would be similar to how `fetchBib` is called on `/all` but would it work to call `fetchBib` and then `fetchItems` with the all flag set to true? Is this partially because of the loading time before the page loads?

Reply:
Doing this for the reason you mentioned, to avoid an extremely long page load time on bibs with a lot items. This will be necessary when JS is disabled, but since almost all users will have it enabled I don't think it makes sense to prevent the rest of the page from rendering.

Reply:
Yes, but we should think of features as "assume users don't have JS (or ship as little JS as possible to the client) and what features can we progressively add". It's a shift from thinking of it as "users won't disable JS", which is generally true, and also aligns with "server-first" approaches.

Reply:
Chatted about this with @EdwinGuzman over slack, we will have a conversation with Vera when she is back on how want to approach JS-disabled detection on the server side.